spec.html

```diff
@@ -35142,28 +35115,75 @@ <h1>Notation</h1>
+              </tr>
+              <tr>
+                <td>[[Multiline]]</td>
+                <td>a Boolean</td>
+                <td>indicates whether *"m"* appears in the RegExp's flags</td>
+              </tr>
+              <tr>
+                <td>[[DotAll]]</td>
+                <td>a Boolean</td>
+                <td>indicates whether *"s"* appears in the RegExp's flags</td>
+              </tr>
+              <tr>
+                <td>[[Unicode]]</td>
+                <td>a Boolean</td>
+                <td>indicates whether *"u"* appears in the RegExp's flags</td>
+              </tr>
+              <tr>
+                <td>[[WordCharacters]]</td>
+                <td>a CharSet</td>
+                <td>the characters considered "word characters" for the purposes of `\\b`, `\\B`, `\\w`, and `\\W`</td>
+              </tr>
+              <tr>
+                <td>[[CapturingGroupsCount]]</td>
+                <td>a non-negative integer</td>
+                <td>the number of left-capturing parentheses in the RegExp's pattern</td>
+              </tr>
+            </table>
+          </emu-table>
+        </emu-clause>
       </emu-clause>
 
       <emu-clause id="sec-compilepattern" type="sdo" oldids="sec-pattern">
-        <h1>Runtime Semantics: CompilePattern ( ): an Abstract Closure that takes a List of characters and a non-negative integer and returns a MatchResult</h1>
+        <h1>
+          Runtime Semantics: CompilePattern (
+            _rer_: a RegExp Record,
+          ): an Abstract Closure that takes a List of characters and a non-negative integer and returns a MatchResult
+        </h1>
         <dl class="header">
         </dl>
         <emu-grammar>Pattern :: Disjunction</emu-grammar>
         <emu-alg>
-          1. Let _m_ be CompileSubpattern of |Disjunction| with argument ~forward~.
-          1. Return a new Abstract Closure with parameters (_inputChars_, _index_) that captures _m_ and performs the following steps when called:
-            1. Assert: _inputChars_ is a List of characters.
-            1. Assert: _index_ is a non-negative integer which is &le; the number of characters in _inputChars_.
-            1. Let _Input_ be _inputChars_. This alias will be used throughout the algorithms in <emu-xref href="#sec-pattern-semantics"></emu-xref>.
-            1. Let _InputLength_ be the number of characters contained in _Input_. This alias will be used throughout the algorithms in <emu-xref href="#sec-pattern-semantics"></emu-xref>.
+          1. Let _m_ be CompileSubpattern of |Disjunction| with arguments _rer_ and ~forward~.
+          1. Return a new Abstract Closure with parameters (_Input_, _index_) that captures _rer_ and _m_ and performs the following steps when called:
+            1. Assert: _Input_ is a List of characters.
+            1. Assert: _index_ is a non-negative integer which is &le; the number of characters in _Input_.
```

Comment:
```suggestion
            1. Assert: _index_ is a non-negative integer which is &le; the number of elements in _Input_.
```

otherwise it might be read as implying there are non-characters in the list

Reply:
I think given the assert on the previous line there's not much risk of that, but I suppose it's also true that there's not much value in repeating "characters" on this line given the previous line.

Reply:
Done. What swayed me was the 4 occurrences of `the number of characters in _Input_` that *aren't* accompanied by an assertion about the nature of `_Input_`.

But note that there are similar ambiguities elsewhere in the spec.